src/redisearch.h

```diff
@@ -102,18 +102,18 @@ typedef struct RSDocumentMetadata_s {
 
   /* Document flags  */
   RSDocumentFlags flags : 8;
+  // Type of source document. Hash or JSON.
+  DocumentType type : 8;
 
+  uint32_t ref_count : 16;
```

Comment:
I agree this number should not get to even more than 3, but if we have a dmd leak it might, maybe we should assert it on DMD_Incref?

Reply:
@MeirShpilraien  Why do we have the ref_count here?
For cursors?

Reply:
I think its a legacy code that was needed when we performed the queries in the background. We need to check if its still needed...